src/cli/index.ts

```diff
@@ -221,11 +225,15 @@ export function inferCLIOptions(opts: Partial<CLIOptions>): CLIOptions {
         srcLang = withDefault<string>(srcLang, "json");
     }
 
+    const lang = opts.lang || inferLang(opts);
+    const language = languageNamed(lang);
+    assert(language !== undefined, `Unsupported output language: ${lang}`);
```

Comment:
Better:

```typescript
if (language === undefined) {
    return panic("blabla");
}
```

Then TS can infer that `language !== undefined`, and the `defined` below isn't necessary anymore.